pkg/controller/deployment/sync.go

```diff
@@ -365,17 +365,6 @@ func (dc *DeploymentController) scale(deployment *extensions.Deployment, newRS *
 		return err
 	}
 
-	// If the new replica set is saturated, old replica sets should be fully scaled down.
-	// This case handles replica set adoption during a saturated new replica set.
-	if deploymentutil.IsSaturated(deployment, newRS) {
-		for _, old := range controller.FilterActiveReplicaSets(oldRSs) {
-			if _, _, err := dc.scaleReplicaSetAndRecordEvent(old, 0, deployment); err != nil {
-				return err
-			}
-		}
-		return nil
-	}
```

Comment:
> I just realized that the real fix for the proportional scaling code is to avoid syncing replica sets with synced replicas annotations which we also need for cases when one of the RS updates fails here.

That change has a corner case that I don't want to solve here so I'll simply update IsSaturated fow now.